### ISSUE TITLE
Go module compatibility

### ISSUE THREAD

**Issue by dahu33** [zricethezav/gitleaks#273]
```sh
% go get -u github.com/zricethezav/gitleaks@v2.1.0                                                                                                                                        
go get github.com/zricethezav/gitleaks@v2.1.0: github.com/zricethezav/gitleaks@v2.1.0: invalid version: module contains a go.mod file, so major version must be compatible: should be v0 or v1, not v2
```

AFAIK, can't have a go.mod with a major version superior to 1 and not following the rule described here: https://github.com/golang/go/wiki/Modules#semantic-import-versioning

**Comment by zricethezav** [comment]
@dahu33 this is resolved in the latest version 3.2.2

**Comment by dahu33** [comment]
awersome, thanks!
